modules/persistent-storage-csi-vsphere-top-aware.adoc

```diff
@@ -0,0 +1,168 @@
+
+. Verify that `CSINode` object has topology keys:
++
+[source, terminal]
+----
+~ $ oc get csinode
+----
++
+Example output:
++
+[source, terminal]
+----
+NAME                        DRIVERS     AGE
+co8-4s88d-infra-2m5vd       1           27m
+co8-4s88d-master-0          1           70m
+co8-4s88d-master-1          1           70m
+co8-4s88d-master-2          1           70m
+co8-4s88d-worker-j2hmg      1           47m
+co8-4s88d-worker-mbb46      1           47m
+co8-4s88d-worker-zlk7d      1           47m
+----
++
+[source, terminal]
+----
+~ $ oc get csinode co8-4s88d-worker-j2hmg -o yaml
+----
++
+Exmaple output:
++
+[source, terminal]
+----
+...
+spec:
+    drivers:
+    - allocatable:
+        count: 59
+    name: csi-vsphere.vmware.com
+    nodeID: co8-4s88d-worker-j2hmg
+    topologyKeys: <1>
+    - topology.csi-vmware.com/openshift-zone
+    - topology.csi-vmware.com/openshift-region
+----
+<1> Topology keys from vSphere `openshift-zone` and `openshift-region` catagory tags.
++
+[NOTE]
+=====
+`CSINode` objects might take some time to receive updated topology information. After the driver is updated, `CSINode` objects should have topology keys in them.
+=====
+
+. Create a storage policy that uses the category tags:
+.. In vCenter, from the main menu, click *Policies and Profiles*.
+.. On the *Policies and Profiles* page, in the left navigation pane, click *VM Storage Policies*.
+.. Click *CREATE*.
+.. Type a name for the storage policy.
+.. For the rules, select tags that target the desired datastore.
+
+. Create a new storage class that uses the new zoned storage policy:
+.. Click *Storage* > *StorageClasses*.
+.. On the *StorageClasses* page, click *Create StorageClass*.
+.. Type a name for the new storage class in *Name*.
+.. Under *Provisioner*, select *csi.vsphere.vmware.com*. 
+.. Under *Additional parameters*, for the StoragePolicyName parameter, set *Value* to the name of the new zoned storage policy that you created earlier.
+.. Click *Create*.
++
+For example:
++
+[source, yaml]
+----
+allowVolumeExpansion: true
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+ name: zoned-sc <1>
+parameters:
+ StoragePolicyName:  zoned-storage-policy <2>
+provisioner: csi.vsphere.vmware.com
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+----
+<1> New topology aware storage class name.
+<2> Specify zoned storage policy.
+
+.Results
+Creating persistent volume claims (PVCs) and persistent volumes (PVs) from the topology aware storage class are truly zonal, and should use the datastore in their respective zone depending on how pods are scheduled.
+
+For example:
+
+[source, terminal]
+----
+~ $ oc get pv <pv-name> -o yaml
+----
+
+[source, terminal]
+----
+...
+nodeAffinity:
+  required:
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: topology.csi.vmware.com/openshift-zone <1>
+        operator: In  
+        values:
+        - openshift-zone
```

Comment:
```suggestion
        - <openshift-zone>
```

This should be the specified tag name, I think <openshift-zone> or <openshift-zone name> should be more clear.

Reply:
I think that we are sticking with those tags. See this comment https://github.com/openshift/openshift-docs/pull/52116#discussion_r1021886744.

Reply:
@lpettyjo Here we need to give a specific tag name for openshift-zone like zone-a/zone-b, so I suggest to use <openshift-zone> or <openshift-zone-name>, cc @gnufied

Reply:
Done.